aiaccel/apps/optimize.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import pickle as pkl
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+from hydra.utils import instantiate
+from omegaconf import OmegaConf as oc  # noqa: N813
+
+from aiaccel.job import AbciJobExecutor
+
+"""
+Usage:
+    python -m aiaccel.apps.optimize objective.sh params.x1="[0, 10]" params.x2="[0, 10]" --config config.yaml
+"""
+
+"""
+config.yaml:
+
+study:
+  _target_: optuna.create_study
+  direction: minimize
+
+  sampler:
+    _target_: optuna.samplers.TPESampler
+    seed: 0
+
+params:
+  _convert_: partial
+  _target_: aiaccel.apps.optimize.HparamsManager
+  x1: [0, 1]
+  x2:
+    _target_: aiaccel.apps.optimize.SuggestFloat
+    name: x2
+    low: 0.0
+    high: 1.0
+    log: false
+
+n_trials: 30
+n_max_jobs: 4
+
+group: gaa50000
+
+"""
+
+
+class HparamsManager:
+    def __init__(self, **params_def: dict[str, float]) -> None:
+        self.params = {}
+        for name, param in params_def.items():
+            if isinstance(param, list):
+                low, high = param
+                self.params[name] = lambda trial, name=name, low=low, high=high: trial.suggest_float(name, low, high)
+            else:
+                self.params[name] = param
+
+    def suggest_hparams(self, trial: Any) -> dict[str, int | float | str] | Exception:
+        hparams = {}
+        for param_name, hp_generator in self.params.items():
+            if callable(hp_generator):
+                hparams[param_name] = hp_generator(trial)
+            else:
+                hparams[param_name] = hp_generator
```

Comment:
先日の意図は，ここを簡略化したい，ということだったので，constantを詰めたい場合は，別途constant用のgenerator書くのが正しい気がします．
```suggestion
        return {name: param_fn(trial) for name, param_fn in self.params.items()}
```